unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ *                      -# Mocking the get_layer_from_id() does return an object
+ *                      -# Calling the controller_screen_remove_layer()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() not be called
+ */
+TEST_F(ControllerTests, controller_screen_remove_layer_success)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviScreen, 1);
+    struct ivi_layout_layer *l_layoutLayer[1] = {(struct ivi_layout_layer *)0xFFFFFFFF};
+    SET_RETURN_SEQ(get_layer_from_id, l_layoutLayer, 1);
+
+    uint32_t l_layer_id = 10;
+    controller_screen_remove_layer(nullptr, nullptr, l_layer_id);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_screenshot_nullScreenShot
+ * @brief               Test case of controller_screen_screenshot() where wl_resource_create() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_screenshot()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_no_memory() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_screenshot_nullScreenShot)
+{
+    uint32_t l_id = 10;
+    controller_screen_screenshot(nullptr, nullptr, nullptr, l_id);
+
+    ASSERT_EQ(wl_resource_post_no_memory_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_no_memory_fake.arg0_history[0], NULL);
+    ASSERT_EQ(wl_resource_get_user_data_fake.return_val_history[0], NULL);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_screenshot_nullScreen
+ * @brief               Test case of controller_screen_screenshot() where wl_resource_create() success, return an object
+ *                      but wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_create() does return an object
+ *                      -# Calling the controller_screen_screenshot()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ *                         +# wl_resource_destroy() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_screenshot_nullScreen)
+{
+    struct wl_resource * screenshot[1] = {(struct wl_resource *)0xFFFFFFFF};
+    SET_RETURN_SEQ(wl_resource_create, screenshot, 1);
+
+    uint32_t l_id = 10;
+    controller_screen_screenshot(nullptr, nullptr, nullptr, l_id);
+
+    ASSERT_EQ(wl_resource_get_user_data_fake.return_val_history[0], NULL);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.arg1_history[0], IVI_SCREENSHOT_ERROR);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 1);
```

Comment:
should verify wl_resource_create_fake.call_count and weston_buffer_from_resource_fake.call_count